File: api/csharp/CloudAPI/Models/DeviceState.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace CloudApi.Models
{
    /// <summary>
    ///   The state of a device, with respect to a game of Button Pong.
    /// </summary>
    /// 
    [JsonConverter(typeof(StringEnumConverter))]
    public enum DeviceState
    {
        /// <summary>The state of the device cannot be determined.</summary>
        Unknown,

        /// <summary>The device was registered for the game, and is still an active participant.</summary>
        RegisteredActive,

        /// <summary>The device was registered for the game, but has been eliminated an is inactive.</summary>
        RegisteredInactive,

        /// <summary>The device was not registered for the game.</summary>
        NotInGame
    }
}

File: api/node/shared/models/deviceState.js
/**
 * The state of a device, with respect to a game of Button Pong.
 * @readonly
 * @enum { string }
 */
const DeviceState = Object.freeze({
    /** TThe state of the device cannot be determined. */
    Unknown : "Unknown",

    /** The device was registered for the game, and is still an active participant. */
    RegisteredActive : "RegisteredActive",

    /** The device was registered for the game, but has been eliminated an is inactive. */
    RegisteredInactive : "RegisteredInactive",

    /** The device was not registered for the game. */
    NotInGame : "NotInGame"
});

module.exports = DeviceState;